nixos/modules/services/hardware/nvidia-container-toolkit/cdi-generate.nix

```diff
@@ -17,11 +17,14 @@
     (mount:
       ["${lib.getExe jq} '${jqAddMountExpression} ${builtins.toJSON (mkMount mount)}'"])
     mounts;
+  ldLibraryPathExport = if builtins.isString nvidia-driver then "export LD_LIBRARY_PATH=${nvidia-driver}/lib" else "";
```

Comment:
Why this behavior only when the provided path is a string? Please explain a little further how this new attribute might be used.

I would like to avoid surprises when `nvidia-driver` is something other than `null` and a string, we will silently fallback to not use the provided parameter. Is there any chance for the user to have provided a nix path?

Also, can we use a flag of `nvidia-ctk cdi generate` rather than using an envvar? I see `--driver-root`; others might be relevant too. I think this is more explicit and lets less room for unexpected behaviors.